Source/CronosPlayUnreal/Public/DynamicContractActor.h

```diff
@@ -0,0 +1,165 @@
+                               FString &output_message);
+
+    /**
+     * EncodeDynamicContract for http access
+     * @param rpcserver    cronos rpc server endpoint
+     * @param contactAddress    contract address
+     * @param abiJson    contract abi json, this is json string, not file path
+     * @param functionName    function name
+     * @param functionArgs    function args in json
+         \code
+     * simple json example: Address, String
+     * solidity function signature: safeMint(address,string)
+     * [{"Address":{"data":"0x00"}},{"Str":{"data":"my"}}]
+         \endcode
+     * @return encoded bytes
+     */
+
+    UFUNCTION(BlueprintCallable,
+              meta = (DisplayName = "EncodeDynamicContract",
+                      Keywords = "Json,Rpc,DynamicContract,Http"),
+              Category = "PlayCppSdk")
+    void EncodeDynamicContract(FString functionName, FString functionArgs,
+                               TArray<uint8> &output, bool &success,
+                               FString &output_message);
+
+    /**
+     * call dynamic contract (non modifying)
+     *
+     * @param rpcserver    cronos rpc server endpoint
+     * @param contactAddress    contract address
+     * @param abiJson    contract abi json, this is json string, not file path
+     * @param functionName    function name
+     * @param functionArgs    function args in json
+         \code
+     * simple json example: Address, String
+     * * solidity function signature: safeMint(address,string)
+     * [{"Address":{"data":"0x00"}},{"Str":{"data":"my"}}]
+     * json example: String
+     * [{"Str":{"data":"test"}}]
+     * [{"Str":{"data":"test"}},{"Int":{"data":"0"}},{"Uint":{"data":"0"}},{"Address":{"data":"0x0000…0000"}},{"FixedBytes":{"data":[0,0,0,0]}},{"Bytes":{"data":[0,0,0,0]}},{"Bool":{"data":false}},{"FixedArray":{"data":[{"Uint":{"data":"0"}},{"Uint":{"data":"0"}}]}},{"Array":{"data":[{"Uint":{"data":"0"}},{"Uint":{"data":"0"}}]}},{"Tuple":{"data":[{"Uint":{"data":"0"}},{"Uint":{"data":"0"}}]}}]
+         \endcode
+     * @return output in json
+     */
+    UFUNCTION(BlueprintCallable,
+              meta = (DisplayName = "CallDynamicContract",
+                      Keywords = "Json,Rpc,DynamicContract,Http"),
+              Category = "PlayCppSdk")
+    void CallDynamicContract(FString functionName, FString functionArgs,
+                             FCallDynamicContractDelegate Out);
+
+    /**
+     * send modifying contract call
+     * @param contractAddress contract address
+     * @param walletindex wallet index which starts from 0
+     * @param abiJson abi json
+     * @param functionName function name
+     * @param functionArgs function args injson
+         \code
+     * simple json example: Address, String
+     * solidity function signature: safeMint(address,string)
+     * [{"Address":{"data":"0x00"}},{"Str":{"data":"my"}}]
+         \endcode
+     * @return  tx-receipt
+     */
```

Comment:
Docs are different than the function signature.

Reply:
it's for sample `safeMint`, i'll clarify